src/log_uploader.js

```diff
@@ -4,48 +4,85 @@ const { dialog, clipboard } = require('electron');
 const { config } = require('./launcher_config');
 const fs = require('fs');
 
-const github = require('octonode');
+const AWS = require('aws-sdk');
+const GistClient = require("gist-client")
 const got = require('got');
 
 const { log, logPath } = require('./spring_log');
 
+const logs_upload_location = config.logs_upload_location;
+const logs_s3_bucket = config.logs_s3_bucket;
+
+const shouldUploadWithS3 = logs_upload_location != null && logs_upload_location == 's3' && logs_s3_bucket != null && logs_s3_bucket != '';
+const shouldUploadWithGithub = logs_upload_location != null && logs_upload_location == 'gist';
+//const shouldUploadWithS3 = logs_upload_location != null && logs_upload_location == 'spring';
+
+function s2b(str) {
+	const bytes = new Uint8Array(str.length / 2);
+	for (let i = 0; i !== bytes.length; i++) {
+		bytes[i] = parseInt(str.substr(i * 2, 2), 16);
+	}
+	return bytes;
+}
+
 // such security, much wow
-const sec = [
-	'ghp',
-	'_j1npS8i',
-	'jzzVLSsJ',
-	'R0NBQVM',
-	'pjUWN1Fi4WdO7n'
+const a1 = [
+	'414B494133',
+	'5843505A41',
+	'51414B424C',
+	'4D52545747'
+];
+
+const a2 = [
+	'6939487543',
+	'386B497255',
+	'673149596A',
+	'48692B6C6A',
+	'33786D734D',
+	'434A5A6D66',
+	'4A4F415875',
+	'7A5543574A'
 ];
-const github_access_token = sec.join('');
```

Comment:
Might want to export these to environment variables or something. I don't mind as it is atm.

Just keep in mind your logs will be mine 😈

Reply:
Yeah, and Gists will be mine... There's really no place we can store it atm, launcher needs to know them when running. So either expose them in config, or fetch from some server